### PR TITLE
FEATURE: added max length setting for text type fields

### PR DESCRIPTION
**Pull request by fzngagan**

### CODE REVIEW

File: lib/custom_wizard/field.rb [hunk 1]
```diff
@@ -3,14 +3,17 @@ def self.types
     @types ||= {
       text: {
         min_length: nil,
+        max_length:nil,
         prefill: nil
       },
       textarea: {
         min_length: nil,
+        max_length:nil,
```
Comment (angusmcleod) [inline lib/custom_wizard/field.rb:11]: @fzngagan Let's get the spacing on this right please.

File: spec/components/custom_wizard/update_validator_spec.rb [hunk 2]
```diff
@@ -49,7 +49,32 @@ def perform_validation(step_id, submission)
...
+    updater = perform_validation('step_1', step_1_field_2: long_string)
+    expect(
+      updater.errors.messages[:step_1_field_2].first
+    ).to eq(I18n.t('wizard.field.too_long', label: 'Textarea', max: max_length))
+
+    updater = perform_validation('step_1', step_1_field_3: long_string)
+    expect(
+      updater.errors.messages[:step_1_field_3].first
+    ).to eq(I18n.t('wizard.field.too_long', label: 'Composer', max: max_length))
+  end
```
Comment (angusmcleod) [inline spec/components/custom_wizard/update_validator_spec.rb:76]: @fzngagan We also need a test of the success case, i.e. when the setting is applied and the input is under the max length.

File: assets/javascripts/discourse/models/custom-wizard.js.es6 [hunk 3]
```diff
@@ -8,7 +8,7 @@ import { popupAjaxError } from 'discourse/lib/ajax-error';
 
 const CustomWizard = EmberObject.extend({
   save(opts) {
-    return new Promise((resolve, reject) => {      
+    return new Promise((resolve, reject) => {
```
Comment (angusmcleod) [inline assets/javascripts/discourse/models/custom-wizard.js.es6:11]: @fzngagan Changes to a file that are just removal of whitespace should not be in the PR

**Review by angusmcleod — CHANGES_REQUESTED** [review]

File: lib/custom_wizard/field.rb (hunk 1 above)
Comment (fzngagan) [reply lib/custom_wizard/field.rb]: Done

File: spec/components/custom_wizard/update_validator_spec.rb (hunk 2 above)
Comment (fzngagan) [reply spec/components/custom_wizard/update_validator_spec.rb:93]: Done

File: assets/javascripts/discourse/models/custom-wizard.js.es6 (hunk 3 above)
Comment (fzngagan) [reply assets/javascripts/discourse/models/custom-wizard.js.es6]: Ok, I've reverted this change from the PR branch. :)

File: spec/components/custom_wizard/update_validator_spec.rb (hunk 2 above)
Comment (angusmcleod) [reply spec/components/custom_wizard/update_validator_spec.rb:93]: @fzngagan This should be seperate tests instead of one big one.

**Pull request merged**